src/inputs/input.jsx

```diff
@@ -218,35 +264,51 @@ class Input extends React.PureComponent {
             onKeyDown,
         } = this.props;
 
-        if (!_.isUndefined(onKeyDown)) {
+        if (isFunction(onKeyDown)) {
             onKeyDown(event);
         }
     }
 
     onNumberToggleClick(action) {
         const {
+            disable,
             disabled,
             max,
             min,
             type,
             onChange,
         } = this.props;
-        const value = this._input.value;
+        const isDisabled = disable || disabled;
+        // eslint-disable-next-line no-underscore-dangle
+        const { value } = this._input;
 
-        if (!disabled) {
-            let newValue = value || 0;
+        if (!isDisabled) {
+            let newValue = value ? toNumber(value) : 0;
 
             switch (action) {
                 case 'down':
-                    newValue = type === 'number' && _.isNumber(min) && value * 1 - 1 < min ? !newValue ? max : newValue : --newValue;
+                    if (type === 'number' && isNumber(min) && newValue - 1 < min) {
+                        if (!newValue) {
+                            newValue = min;
+                        }
+                    } else {
+                        newValue -= 1;
+                    }
+
                     break;
                 case 'up':
-                    newValue = type === 'number' && _.isNumber(max) && value * 1 + 1 > max ? newValue : newValue < min ? min : ++newValue;
+                    if (type === 'number' && isNumber(max) && newValue + 1 > max) {
+                        newValue = max;
+                    } else {
+                        newValue += 1;
+                    }
+
                     break;
                 default:
             }
 
-            if (_.isUndefined(onChange)) {
+            if (!isFunction(onChange)) {
+                // eslint-disable-next-line no-underscore-dangle
                 this._input.value = newValue;
             } else {
                 onChange(newValue);
```

Comment:
Shouldn't we use your new `setNewValue()` method for this?

Reply:
Fixing...

Reply:
Commit 7c8b691 fixes this.

Reply:
Yep, looks good.  Resolving.